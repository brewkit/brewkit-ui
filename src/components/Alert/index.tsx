import * as React from 'react';
import MuiAlert from '@material-ui/lab/Alert';
import MuiAlertTitle from '@material-ui/lab/AlertTitle';
import { Props } from './types';


/**
 * An alert displays a short, important message in a way that attracts the user's attention without interrupting the
 * user's task.
 *
 * [Material-UI Docs](https://material-ui.com/components/alert/)
 *
 * ## Differences in Brewkit:
 *
 * - Adds a `title` prop that can be used in place of `<AlertTitle />`.
 */
export const Alert = React.forwardRef(({
    title,
    children,
    ...otherProps
}: Props, ref: React.Ref<any>): React.ReactElement => (


    <MuiAlert ref={ref} {...otherProps}>
        {title && <MuiAlertTitle>{title}</MuiAlertTitle>}
        {children}
    </MuiAlert>


));


Alert.displayName = 'Alert';


export default Alert;
