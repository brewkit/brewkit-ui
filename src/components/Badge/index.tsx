import React, { ReactElement } from 'react';
import clsx from 'clsx';
import { Props } from './types';


/**
 * The Badge component is used to attach content to it's child or can be used inline with no children
 */
function Badge({
    anchorFrom = 'top-right',
    children,
    className,
    color = 'info',
    content,
    isCircular = false,
    ...otherProps
}: Props): ReactElement {


    const badgeClasses = clsx(
        'brew-Badge',
        `brew-Badge--anchorFrom-${anchorFrom}`,
        `brew-Badge--color-${color}`,
        { 'brew-Badge--isCircular': isCircular },
        { 'brew-Badge--inline': !children },
        className,
    );


    return (
        <div className={badgeClasses}>
            {children}
            <span className="brew-Badge__element" {...otherProps}>{content}</span>
        </div>
    );


}


export default Badge;
