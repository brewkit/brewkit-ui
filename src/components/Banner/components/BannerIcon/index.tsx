import React, { ReactElement } from 'react';
import clsx from 'clsx';
import Icon from '../../../Icon';
import Progress from '../../../Progress';
import { Props } from './types';


function BannerIcon({
    className,
    iconName = '',
    isLoading,
    ...otherProps
}: Props): ReactElement {


    const iconClasses = clsx(
        'brew-Banner__icon',
        className,
    );


    const loading = (
        <Progress className="brew-Banner__iconLoading" size="medium" variant="circular" />
    );


    const icon = (
        <Icon className="brew-Banner__iconImage" size="large">
            {iconName}
        </Icon>
    );


    return (
        <div className={iconClasses} {...otherProps}>
            {isLoading ? loading : icon}
        </div>
    );


}


export default BannerIcon;
