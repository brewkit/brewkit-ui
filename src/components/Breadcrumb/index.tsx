import React, { ReactElement } from 'react';
import { BrowserRouter, Link } from 'react-router-dom';
import clsx from 'clsx';
import Typography from '../Typography';
import { Props, Crumb } from './types';


function Breadcrumb({
    className,
    divider,
    crumbs,
}: Props): ReactElement {


    const crumbCount = crumbs.length - 1;
    const classes = clsx(
        'brew-Breadcrumb',
        className,
    );


    return (
        <div className={classes}>

            {crumbs.map((crumb: Crumb, index: number) => {
                const notLast = crumbCount !== index;
                return (
                    <BrowserRouter key={index}>
                        <Typography variant="body2">
                            {crumb.url && notLast
                                ? (
                                    <Link className="brew-Breadcrumb__link" to={crumb.url}>
                                        {crumb.title}
                                    </Link>
                                )
                                : (
                                    <Typography variant="body2">
                                        {crumb.title}
                                    </Typography>
                                )
                            }
                            <span className="brew-Breadcrumb__separator">
                                {divider
                                    ? notLast && divider
                                    : notLast && '/'
                                }
                            </span>
                        </Typography>
                    </BrowserRouter>
                );
            })}

        </div>
    );


}


export default Breadcrumb;
