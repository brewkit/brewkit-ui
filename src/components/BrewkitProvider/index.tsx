import React from 'react';
import { StylesProvider, createGenerateClassName } from '@material-ui/core/styles';
import { Props } from './types';


const generateClassName = createGenerateClassName({
    productionPrefix: 'bk',
    disableGlobal: true,
});


/**
 * The `<BrewkitProvider />` component is intended to wrap your Brewkit application and helps to simplify configuration.
 */
export const BrewkitProvider = ({
    children,
}: Props): React.ReactElement => (

    <StylesProvider generateClassName={generateClassName} injectFirst>
        {children}
    </StylesProvider>

);


BrewkitProvider.displayName = 'BrewkitProvider';


export default BrewkitProvider;

