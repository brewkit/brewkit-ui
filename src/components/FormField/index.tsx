import * as React from 'react';
import { useFormContext } from 'react-hook-form';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import { AnimatePresence, motion } from 'framer-motion';
import Checkbox from '@components/Checkbox';
import Radio from '@components/Radio';
import TextField from '@components/TextField';
import Switch from '@components/Switch';
import { Props } from './types';


/* Map our input components for easier access */
const components = {
    checkbox: Checkbox,
    radio: Radio,
    switch: Switch,
    text: TextField,
    password: TextField,
    number: TextField,
};


/**
 * `FormField` is an abstraction of most inputs, additionally binding the rendered input to a `Form` so they can be
 * tracked and validated.
 *
 * > This component is not found in Material UI.
 */
export const FormField = React.forwardRef(({
    type = 'text',
    label,
    name,
    validation = {},
    helperText,
    ...otherProps
}: Props, ref: React.Ref<any>): React.ReactElement => {


    const { register, errors } = useFormContext();
    const Component: any = components[type] || TextField;


    /**
     * configure our Framer animation
     */
    const fadeAnim = {
        initial: { opacity: 0, y: -5 },
        animate: { opacity: 1, y: -0 },
        exit: { opacity: 0 },
    };


    /**
     * determine what to render in the helper text. Keys have to be changed based on content ot ensure the component
     * updates correctly.
     */
    function getHelperText(): React.ReactNode {

        let content: React.ReactNode = ' ';
        let key = 1;

        /** if there is an error, return that */
        if (errors[name]?.message) {
            content = errors[name]?.message;
            key = 2;
        }

        /** else, return helperText if provided */
        if (helperText) {
            content = helperText;
            key = 3;
        }

        if (key === 1) return <AnimatePresence />;

        return (
            <AnimatePresence>
                <motion.span key={key} layout style={{ display: 'inline-block' }} {...fadeAnim}>
                    {content}
                </motion.span>
            </AnimatePresence>
        );

    }


    /**
     * if the component is a TextField, we just use that.
     */
    if (Component === TextField) return (
        <Component
            error={Boolean(errors[name])}
            helperText={getHelperText()}
            inputRef={register(validation)}
            label={label}
            name={name}
            type={type}
            {...otherProps}
        />
    );


    /**
     * otherwise, we wrap it in a FormControlLabel
     */
    return (
        <FormControlLabel
            control={<Component name={name} {...otherProps} />}
            inputRef={register(validation)}
            label={label}
        />
    );


});


FormField.displayName = 'FormField';


export default FormField;
