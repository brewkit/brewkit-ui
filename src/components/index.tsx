import * as Mui from '@material-ui/core';


export { default as BrewkitProvider } from './BrewkitProvider';
export { default as Button } from './Button';
export { default as Flag } from './Flag';
export { default as Icon } from './Icon';
export { default as Input } from './Input';
export { default as Progress } from './Progress';
export { default as Tooltip } from './Tooltip';
export { default as Typography } from './Typography';


/**
 * Re-export MUI for easy use
 */
export { Mui };
