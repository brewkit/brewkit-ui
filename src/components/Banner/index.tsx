import React, { ReactElement } from 'react';
import clsx from 'clsx';
import BannerIcon from './components/BannerIcon';
import BannerBody from './components/BannerBody';
import BannerButton from './components/BannerButton';
import BannerInfo from './components/BannerInfo';
import BannerTitle from './components/BannerTitle';
import { Props } from './types';


function Banner({
    children,
    className,
    color,
    iconName,
    description,
    isLoading,
    title,
    variant,
    ...otherProps
}: Props): ReactElement {


    const bannerClasses = clsx(
        'brew-Banner',
        `brew-Banner--color-${color}`,
        `brew-Banner--${variant}`,
        className,
    );


    return (
        <div className={bannerClasses} {...otherProps}>
            {iconName && (
                <BannerIcon iconName={iconName} isLoading={isLoading} />
            )}
            {(title || description) && (
                <BannerBody>
                    {title && (
                        <BannerTitle>{title}</BannerTitle>
                    )}
                    {description && (
                        <BannerInfo>{description}</BannerInfo>
                    )}
                </BannerBody>
            )}
            {children && (
                <div className="brew-Banner__content">{children}</div>
            )}
        </div>
    );


}


Banner.Body = BannerBody;
Banner.Button = BannerButton;
Banner.Icon = BannerIcon;
Banner.Info = BannerInfo;
Banner.Title = BannerTitle;


export default Banner;
