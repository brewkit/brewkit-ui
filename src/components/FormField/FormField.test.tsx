// @ts-nocheck
import * as React from 'react';
import { create } from 'react-test-renderer';
import { FormControlLabel } from '@material-ui/core';
import components from './componentTypes';

it('Renders TextField component', () => {
    const TextInput = components.text;
    const comp = create(
        <FormControlLabel
            control={<TextInput name="textinput" />}
            label={<label>Some label</label>}
        />
    ).toJSON();

    expect(comp).toMatchSnapshot();
});

it('Renders Radio component', () => {
    const Radio = components.radio;
    const comp = create(
        <FormControlLabel
            control={<Radio name="radio" checked />}
            label={<label>Some label</label>}
        />
    ).toJSON();

    expect(comp).toMatchSnapshot();
});

it('Renders Switch component', () => {
    const Switch = components.switch;
    const comp = create(
        <FormControlLabel
            control={<Switch name="switch" checked />}
            label={<label>Some label</label>}
        />
    ).toJSON();

    expect(comp).toMatchSnapshot();
});

it('Renders Checkbox component', () => {
    const Checkbox = components.checkbox;
    const comp = create(
        <FormControlLabel
            control={<Checkbox name="checkbox" checked />}
            label={<label>Some label</label>}
        />
    ).toJSON();

    expect(comp).toMatchSnapshot();
});
