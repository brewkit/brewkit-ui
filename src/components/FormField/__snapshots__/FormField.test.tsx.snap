// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Checkbox component 1`] = `
<label
  className="MuiFormControlLabel-root"
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-7 Mui-checked MuiIconButton-colorSecondary"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={true}
        className="PrivateSwitchBase-input-9"
        data-indeterminate={false}
        name="checkbox"
        onChange={[Function]}
        type="checkbox"
      />
      <div
        className="Checkbox-controlBase-13 Checkbox-checkedControl-15"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          opacity={0}
          style={Object {}}
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
      </div>
    </span>
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  >
    <label>
      Some label
    </label>
  </span>
</label>
`;

exports[`Renders Radio component 1`] = `
<label
  className="MuiFormControlLabel-root"
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-7 Mui-checked MuiIconButton-colorSecondary"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={true}
        className="PrivateSwitchBase-input-9"
        name="radio"
        onChange={[Function]}
        type="radio"
      />
      <div
        className="PrivateRadioButtonIcon-root-10 PrivateRadioButtonIcon-checked-12"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-11"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </div>
    </span>
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  >
    <label>
      Some label
    </label>
  </span>
</label>
`;

exports[`Renders Switch component 1`] = `
<label
  className="MuiFormControlLabel-root"
>
  <span
    className="MuiSwitch-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-7 Mui-checked"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          checked={true}
          className="PrivateSwitchBase-input-9 MuiSwitch-input"
          name="switch"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="MuiSwitch-thumb"
        />
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </span>
    <span
      className="MuiSwitch-track"
    />
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  >
    <label>
      Some label
    </label>
  </span>
</label>
`;

exports[`Renders TextField component 1`] = `
<label
  className="MuiFormControlLabel-root"
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        name="textinput"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  >
    <label>
      Some label
    </label>
  </span>
</label>
`;
